### ISSUE TITLE
Remove Discord’s verified status

### ISSUE THREAD

**Issue by Eonfge** [flathub/com.discordapp.Discord#343]
Over on the Flathub Discource, I started a debate related to the 'verified' status of commercial packages who don't get any upstream support or attention... In other words, it's about this Discord package.

https://discourse.flathub.org/t/make-the-verified-status-more-strict-i-e-remove-discords-verified-status/5385

I welcome you all to join the discussion

**Comment by cassidyjames** [comment]
I commented a bit more verbosely on the thread, but I'll leave some shorter thoughts here.

Discord have **just** actively taken an interest in working on and improving the Flatpak. Transitioning a community-submitted and -maintained manifest for a proprietary app to something that Discord themselves want to get behind and promote has some growing pains, but we've _just_ taken our first steps. I think it's wildly premature—and would greatly damage our image and relationship with ISVs—to just turn around and say "remove verification because we still have work to do."

For example, you mention not being able to integrate certain library fixes. If this Flatpak has their blessing and is ultimately under their control, that should remove that barrier. Longer-term the goal should be to have Discord build the Flatpak out as part of their regular build and release process; they now have a space where they can move in that direction since they have ownership over their app ID. I know several Discord employees are already using the Flatpak, and they’re excited to move forward to improve it; let's work together on a way forward instead of backwards.

**Comment by guihkx** [comment]
Is this still a relevant subject these days?

I became a maintainer of this package a few days ago, and I haven’t seen any Discord employees working directly on it.

However, I did notice that Discord has been releasing stable builds for Linux much more frequently in recent times.

In fact, their latest stable release (0.0.70) already includes Electron 32.0.0, which is also the latest stable release they have. This is something I never saw happen in the past. Discord were always many versions behind.

And given that 99% of the bugs reported here were actually caused by the ancient Electron version they were shipping, I'd say that's a pretty significant improvement.

**Comment by lionirdeadman** [comment]
Personally speaking, I don't feel great about the verified status.

It personally made me feel more uneasy about working on the package as I felt that changing things would now reflect on Discord's image and I did not and do not have any paths to actually collaborate with any Discord developers.

While there seems to have been an intention from Discord to work on the flatpak, it seems to have never materialized. I don't really like the idea that Discord can pass off volunteer work as their own and I don't like the idea that people think Discord developers actually have input on this package's maintenance (which they clearly do not).

**Comment by WindFreaker** [comment]
The issue that has broken the file chooser portal proves Discord has no intention in giving the flatpak version proper treatment or care. It should not be promoted as if it is verified / official. This wrongfully signals to users that

1. Linux and flatpak have first-party support and are a priority to Discord Inc.
2. The app is ensured to be safe by Discord's own standards.

The work the maintainers put into this project is admirable, but the bottom line is that badge signals that this isn't a community effort and instead coming from the direct source. That simply is not true.

**Comment by Beryesa** [comment]
Well, tbf it was electron that broke "production" code which these companies generally use blindly.
Sure discord was inconsiderate rolling with that, but that's an upstream issue nonetheless.

**Comment by Eonfge** [comment]
I think my original points still hold true: The package here is maintained by volunteers like @guihkx and while I really appreciate their effort, that's not what verified means.

There is also a problem with commitment: They want an official package, but they don't want to do any work testing. If it works, it's their achievement and when it fails, it's the communities' failure. Discord wants their cake and eat it to.

So yeah, this package should not be verified.

**Comment by SilverLPs** [comment]
Honestly, I hate to say it, but @Eonfge is right. The flatpaks verified icon should be removed asap in our all interest. It is misleading and leads to confusion and higher expectations than this project is able to satisfy (meaning you don't have the option to influence the development of Discord at all, nobody besides Discord has that). 

I absolutely appreciate the work you are doing here for free for everyone of us. But you can not gurantee for the quality of this package, as most of the Software depends on Discord and can not be influenced by anyone else. 

It's probably just a matter of time until the next bug appears that would make another freeze of this version necessary until the problem gets eventually fixed much later by coincidence or something out of own control.

**Comment by nolddor** [comment]
Completely agree with this GitHub issue. If the Discord team is not involved in or actively supporting this project, I don't see the relevance of the verified icon. This doesn't mean that the project isn't excellent or that your hard work isn't widely appreciated. However, the icon can be misleading, for majority of people.

**Comment by Eonfge** [comment]
@barthalion Is this something that could be reviewed in the Admin team? @guihkx is doing a great job but he is no Discord employee

**Comment by cassidyjames** [comment]
The Discord team is continuing to actively engage with us; for example, they flagged the audio issue, reported it as an [incident] on their status site, and reached out to roll it back until they can ship a proper fix.

A reminder of what verification means, from the [docs]:

>A verified app on Flathub is one whose developer has confirmed their ownership of the app ID using a uniquely generated token. This usually also may mean that either the app is maintained directly by the developer or a party authorised or approved by them.

Discord fits this definition: they own the app ID and have confirmed it by adding the Flathub token to their domain. They have also authorized the maintenance and distribution of the app via this repo, ultimately sourced from their official Linux tarballs.

I'm continuing to work to get someone officially representing Discord interacting directly on this repo; this latest incident may actually serve as an opportunity there. But regardless, Discord fits the definition of a verified app and we shouldn't strip that away unless we first change the definition and policies around verification.

[incident]: https://discordstatus.com/incidents/mv9k3px764kd
[docs]: https://docs.flathub.org/docs/for-users/verification

**Comment by lionirdeadman** [comment]
> I'm continuing to work to get someone officially representing Discord interacting directly on this repo; this latest incident may actually serve as an opportunity there. But regardless, Discord fits the definition of a verified app and we shouldn't strip that away unless we first change the definition and policies around verification.

As I've been away, I know you've been in contact with Discord, but is this contact point also shared with @guihkx? I feel this would be valuable. It personally felt strange when I was the main maintainer of this package and I had radio silence. 

It would help in the event that you need to step away from the project (bus factor and all). I think it would also generally feel better if the package maintainer would see the effort being put in by Discord rather than it being communicated by another community member.

I know that I personally felt that no effort was being put while I was maintainer and I think that feeling is also likely born from the fact I never received any input from Discord whatsoever.

I'd like your thoughts and @guihkx's.

**Comment by guihkx** [comment]
I don't have their contact information (not that I ever asked for it or expected to receive it).

I'll be honest: I don’t value the verified badge as much as others seem to. Yes, it would be great if a Discord employee could maintain this package, but my understanding is that the original agreement between Discord's developers and the Flathub collaborators was to entrust maintenance to the community without direct involvement (please correct me if I'm wrong).

In short: I don't mind the status quo, nor do I mind if the verification badge is removed. Either way, I'll do my best to maintain this package until I lose interest.

**Comment by SilverLPs** [comment]
> until I lose interest.

And that is another good example why the badge should be removed.

**Comment by lionirdeadman** [comment]
> > until I lose interest.
> 
> And that is another good example why the badge should be removed. (Meaning there is no gurantee for future updates, as the package is not an official one...)

There has never been and never will be guarantees for maintainership whether that person is employed by Discord inc. or not. I don't think this is an argument at all.

**Comment by SilverLPs** [comment]
1. There is ofc no full gurantee just because it is maintained officially by Discord, however if a package is linked to the actual developer of the software it is just way more unlikely to get abolished compared to maintainership that is dependant on third party community effort, often just single persons doing it in their free time. . 
This situation could theoretically be improved by having bigger communities or organizations that would maintain a larger number of applications, like linuxserver.io does for Docker. Two different badges, one for official images and one for "approved/verified reliable third party maintainers" could further improve on this. However that is just theory and would need to be implemented by Flathub I guess.

2. If a verified (badge) package gets abolished or deprecates I would usually expect that the actual developer of the software is responsible. If it has a badge and is maintained by third party this could lead to wrong assumptions/conclusions. One could argue that Discord is responsible as soon as they manage to give the Flathub package the opportunity to get the badge but thats not helpful when it comes to making Flathub more transparent in practice.

**Comment by alextpedro** [comment]
I am chiming in as a Discord user who moved from the .deb to the Flatpak due to ease of updating.

When I did so, I assumed the Flatpak was supported by Discord themselves and the reason it wasn't mentioned on the website was because engineering and marketing aren't always talking properly to one another. 

Thinking that the Flatpak was official, I had a rather frustrating interaction with Discord support in which they informed me that I wasn't running Discord in an officially supported platform. Naturally, since the Flatpak is verified, I assumed the support agent was misinformed, as they are often external contractors. 

In light of that, I would caution against maintaining Discord's verified status, as it is entirely misleading.

**Comment by Eonfge** [comment]
> Thinking that the Flatpak was official, I had a rather frustrating interaction with Discord support in which they informed me that I wasn't running Discord in an officially supported platform. Naturally, since the Flatpak is verified, I assumed the support agent was misinformed, as they are often external contractors.

Well, how much more convincing do you people need? The Discord developers want their cake and eating it too, and Flathub users are paying the price for that. Remove the verified status and only reinstate it when Discord employees themselves actually maintain this application.

**Comment by lionirdeadman** [comment]
> I'd assume Verified means that the official makers of the application are publishing and updating the application to Flathub, e.g.: Firefox is fully managed by Mozilla, basically every verified Libadwaita application is published and updated to Flathub by their creator, etc.

This assumption would be wrong and for most open source programs, the 'official makers' is a weird concept honestly. This is what the [Flatpak docs](https://docs.flathub.org/docs/for-app-authors/verification) say:
> Verification is the process by which Flathub and developers confirm that an app is published by the original developer or an authorized party.

> If Discord isn't being published and updated to Flathub by _Discord_, that effectively represents Flathub going against what they've otherwise established Verified means, and at worst grounds for a defamation case, for lack of better lawsuit type coming to my head, if the Flatpak breaks in a significant way the official package by Discord doesn't - like others have said here, users will think Discord is 100% making the Flatpak, through no fault of their own, and could bombard Discord about issues the Flatpak could exclusively cause in the future... and _that_ could even equally risk shattering the relationship established with Discord while it's still as brittle as it currently is.

There's no code in this repository that controls this status. Discord themselves verified the application just like any other application would do so. [This file](https://discord.com/.well-known/org.flathub.VerifiedApps.txt) is what proves that Discord inc. has made this effort as the Flathub docs say they should if they want to verify the application. I see no ill play on Flathub on Discord's part so saying that either party is being disingenuous would be weird.

> The Discord developers want their cake and eating it too, and Flathub users are paying the price for that. Remove the verified status and only reinstate it when Discord employees themselves actually maintain this application.

I'm not aware of Flathub removing verified status when an application fullfils the well-known requirements. I think this would be an odd precedent to set especially when there has been no issue of note to remove it other than some people think Discord employees should be the ones maintaining the application on Flathub rather than an authorized party (which I partly support).

For all intents and purposes, removing the application's verified status would be an ideological decision rather than one based on any Flathub documentation, process or other pre-existing position. It would be quite literally unprecedented behaviour.

**Comment by mZorski** [comment]
preface: Don't get me wrong, I'm very grateful for all the work that has been put into this.

> This assumption would be wrong and for most open source programs, the 'official makers' is a weird concept honestly. This is what the Flatpak docs say: [...]

Flatpak's definition of verified is irrelevant, if general assumption of application being verified is that it's published and maintained by its creator. Otherwise this status is worthless.
I don't see why it would be a weird concept for most open source programs. You are either creator of given application, or your work is derived from someone else's and you do not pretend to be original creator.



> There's no code in this repository that controls this status. Discord themselves verified the application just like any other application would do so. This file is what proves that Discord inc. has made this effort as the Flathub docs say they should if they want to verify the application. I see no ill play on Flathub on Discord's part so saying that either party is being disingenuous would be weird.

It doesn't matter that at some point in time Discord verified this application, as reading initial comments, tells me that at some point they had plans for Flatpak support, but dropped it. Even them having plans for that is no ground to make it verified, until they actually take ownership/start maintaining it/officially endorse it/take responsibility if things go south.



> I think this would be an odd precedent to set especially when there has been no issue of note to remove it other than some people think Discord employees should be the ones maintaining the application on Flathub rather than an authorized party (which I partly support).

It would not be "an odd precedent", as it would make it clear for people that it's not official Discord application. Let's not act like default assumption of being verified means something completely different. Ask any everyday Joe what he thinks verified means in this case, and I guarantee that most (if not all) will answer that it's Discord's official app.



> For all intents and purposes, removing the application's verified status would be an ideological decision rather than one based on any Flathub documentation, process or other pre-existing position. It would be quite literally unprecedented behaviour.

Removing it is to stop misleading users, to believe it's official Discord's application, instead of something created by a 3rd party with no official support. Most people will not read documentation, just to verify if something fairly obvious, like verified status, means something completely different, because Flaptak decided so, nor will they dive into the code.

**Comment by Erick555** [comment]
I think you should raise this in some discord support channel and ask them to remove https://discord.com/.well-known/org.flathub.VerifiedApps.txt 

If discord verified this then only they can _unverify_ it.
